Sources/SKCore/CompilationDatabase.swift

```diff
@@ -70,11 +71,19 @@ public protocol CompilationDatabase {
 /// Loads the compilation database located in `directory`, if any.
 public func tryLoadCompilationDatabase(
   directory: AbsolutePath,
+  searchPaths: [RelativePath],
   _ fileSystem: FileSystem = localFileSystem
 ) -> CompilationDatabase? {
-  return
-    (try? JSONCompilationDatabase(directory: directory, fileSystem))
-    ?? (try? FixedCompilationDatabase(directory: directory, fileSystem))
+  return searchPaths
+    .lazy
+    .map { directory.appending($0) }
+    .compactMap {
+      try?
+        (JSONCompilationDatabase(directory: $0, fileSystem)
+        ?? FixedCompilationDatabase(directory: $0, fileSystem))
```

Comment:
If the initializers now return `nil` when the file doesn’t exist, I think we no longer need to do `try?` here and can change it to `try`. That way we don’t silently ignore errors anymore.

Reply:
I looked into this but it is actually a bigger change than it seems as the `throws` will propagate all the way up to things like the `BuildSystem` protocol, so you may want to do that separately (if at all). In the meantime I made it `try!` since decoding errors should be very infrequent and shouldn't fail silently.